cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -0,0 +1,119 @@
+   */
+  @GET
+  @Path("/namespaces/{namespace-id}/profiles")
+  public void getProfiles(HttpRequest request, HttpResponder responder,
+                          @PathParam("namespace-id") String namespaceId,
+                          @Nullable @QueryParam("includeSystem") String includeSystem) {
+    // TODO: implement the method
+  }
+
+  /**
+   * Get the information about a specific profile.
+   */
+  @GET
+  @Path("/namespaces/{namespace-id}/profiles/{profile-name}")
+  public void getProfile(HttpRequest request, HttpResponder responder,
+                         @PathParam("namespace-id") String namespaceId,
+                         @PathParam("name") String profileName) throws NotFoundException {
+    // TODO: implement the method
+  }
+
+  /**
+   * Write a profile in a namespace.
+   */
+  @PUT
+  @Path("/namespaces/{namespace-id}/profiles/{profile-name}")
+  public void writeProfile(HttpRequest request, HttpResponder responder,
+                           @PathParam("namespace-id") String namespaceId,
+                           @PathParam("name") String profileName) throws BadRequestException {
+    // TODO: implement the method
+  }
+
+  /**
+   * Delete a profile from a namespace. A profile must be in the disabled state before it can be deleted.
+   * Before a profile can be deleted, it cannot be assigned to any program or schedule,
+   * and it cannot be in use by any running program.
+   */
+  @DELETE
+  @Path("/namespaces/{namespace-id}/profiles/{profile-name}")
+  public void deleteProfile(HttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId,
+                            @PathParam("name") String profileName) throws NotFoundException, ConflictException {
+    // TODO: implement the method
+  }
+
+  /**
+   * Disable the profile, so that no new program runs can use it,
+   * and no new schedules/programs can be assigned to it.
+   */
+  @POST
+  @Path("/namespaces/{namespace-id}/profiles/{profile-name}/disable")
+  public void disableProfile(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespaceId,
+                             @PathParam("name") String profileName) throws NotFoundException {
+    // TODO: implement the method
+  }
+
+  /**
+   * Enable the profile, so that programs/schedules can be assigned to it.
+   */
+  @POST
+  @Path("/namespaces/{namespace-id}/profiles/{profile-name}/disable")
+  public void enableProfile(HttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId,
+                            @PathParam("name") String profileName) throws NotFoundException {
+    // TODO: implement the method
+  }
+
+  /**
+   * List runs that the profile has been assigned to, either as the manual run profile or as a schedule profile.
+   */
+  @POST
+  @Path("/namespaces/{namespace-id}/profiles/{profile-name}/programs")
```

Comment:
we're removing this endpoint and using metadata instead. I forgot to remove it from the wiki, but have removed it now.